test/server/test_image_validation.py

```diff
@@ -25,59 +19,103 @@ def user_setup():
-            break
-
-        resp = user.session.put(
-            shot_data,
-            json=shot_json,
-        )
-        print(resp.text)
-        assert resp.status_code == 500
-
-
-def test_invalid_data_image_decoded():
-    image = image_setup()
+    invalid_header = 'data:image/png;base64,R0k'
+    assert valid_header in image
+
+    for clip_id in shot_json['clips']:
+        shot_json['clips'][clip_id]['image']['url'] = invalid_header
+        break
+
+    resp = user.session.put(shot_data, json=shot_json,)
+    print(resp.text)
+    assert resp.status_code == 500
+
+
+def test_invalid_png_data_image_decoded():
+    image = example_images['example_images'][0]['url']
     (shot_data, shot_json, user) = user_setup()
-    if "iVBORw0KGgo" in image:
-        invalid_data_image = image.replace('iVBORw0KGgo', 'someIM4gEgo')
-        for clip_id in shot_json['clips']:
-            shot_json['clips'][clip_id]['image']['url'] = invalid_data_image
-            break
-
-        resp = user.session.put(
-            shot_data,
-            json=shot_json,
-        )
-        print(resp.text)
-        assert resp.status_code == 500
-
-
-def test_invalid_data_url():
-    image = image_setup()
+    assert "iVBORw0KGgo" in image
+
+    invalid_data_image = image.replace('iVBORw0KGgo', 'someIM4gEgo')
+
+    for clip_id in shot_json['clips']:
+        shot_json['clips'][clip_id]['image']['url'] = invalid_data_image
+        break
+
+    resp = user.session.put(shot_data, json=shot_json,)
+    print(resp.text)
+    assert resp.status_code == 500
+
+
+def test_invalid_png_data_url():
+    image = example_images['example_images'][0]['url']
     (shot_data, shot_json, user) = user_setup()
-    if "data:image/png;base64" in image:
-        invalid_data_url = image.replace('data:image/png', 'data:image/foo')
-        for clip_id in shot_json['clips']:
-            shot_json['clips'][clip_id]['image']['url'] = invalid_data_url
-            break
+    assert "data:image/png;base64" in image
+    invalid_data_url = image.replace('data:image/png', 'data:image/foo')
+
+    for clip_id in shot_json['clips']:
+        shot_json['clips'][clip_id]['image']['url'] = invalid_data_url
+        break
+
+    resp = user.session.put(shot_data, json=shot_json,)
+    print(resp.text)
+    assert resp.status_code == 500
+
+
+def test_invalid_jpeg_data_image():
+    image = example_images['example_images'][3]['url']
+    (shot_data, shot_json, user) = user_setup()
+    valid_header = "data:image/jpeg;base64,/9j/2wBDAAQDAwQDAwQEAwQFBAQ"
+    invalid_header = "data:image/jpeg;base64,BAAQ"
+    assert valid_header in image
+
+    for clip_id in shot_json['clips']:
+        shot_json['clips'][clip_id]['image']['url'] = invalid_header
+        break
+
+    resp = user.session.put(shot_data, json=shot_json,)
+    print(resp.text)
+    assert resp.status_code == 500
+
+
+def test_invalid_data_jpeg_image_decoded():
```

Comment:
This test is failing.

It returns HTTP status code 200 when we want 500. The original version works though, so we can switch back to that.